src/openMVG/features/sift/SIFT_popSIFT_describer.cpp

```diff
@@ -0,0 +1,58 @@
+#include "SIFT_popSIFT_describer.hpp"
+#include <openMVG/logger.hpp>
+
+namespace openMVG {
+namespace features {
+
+bool SIFT_popSIFT_ImageDescriber::Describe(const image::Image<unsigned char>& image,
+                                      std::unique_ptr<Regions>& regions,
+                                      const image::Image<unsigned char>* mask)
+{
+  popsift::cuda::device_prop_t deviceInfo;
```

Comment:
@griwodz 
In the RT camera tracker, we first create all imageDescribers and we call the `Describe` method in a dedicated thread per camera. We have one thread per camera, but these threads are generated by tbb at each frame.
So we put the call to `cudaSetDevice(_cudaPipeIndex)` (which is done inside `popsift::cuda::device_prop_t::set`) in the `Describe` function because cudaSetDevice relies on the calling host thread.

Currently the code works fine with one camera. But if we use multiple cameras, we get : 

```
Choosing device Choosing device 00: GeForce GTX 980 Ti: GeForce GTX 980 Ti

Device information:Device information:
    Name: GeForce GTX 980 Ti
    Compute Capability:    5
    Name: GeForce GTX 980 Ti
    Compute Capability:    5.2
    Total device mem:      .2
    Total device mem:      6372589568 B 6223232 kB 6077 MB 
    Per-block shared mem:  49152
    Warp size:             32
    Max threads per block: 1024
    Max threads per SM(X): 2048
    Max block sizes:       {1024,1024,64}
    Max grid sizes:        {2147483647,65535,65535}
    Number of SM(x)s:      22
    Concurrent kernels:    yes
    Mapping host memory:   yes
    Unified addressing:    yes

6372589568 B 6223232 kB 6077 MB 
    Per-block shared mem:  49152
    Warp size:             32
    Max threads per block: 1024
    Max threads per SM(X): 2048
    Max block sizes:       {1024,1024,64}
    Max grid sizes:        {2147483647,65535,65535}
    Number of SM(x)s:      22
    Concurrent kernels:    yes
    Mapping host memory:   yes
    Unified addressing:    yes

/s/apps/packages/mikrosVfx/multiview/popsift/develop/repo/src/popsift/sift_desc.cu:/s/apps/packages/mikrosVfx/multiview/popsift/develop/repo/src/popsift/sift_extremum.cu:8370 Runtime error:
    Failed to (re)allocate memory for downloading -1815989562 descriptors

    cudaEventRecord failed: an illegal memory access was encountered
    Not enough memory.
```

Do you have an idea how we should use it?